docker-entrypoint.sh

```diff
@@ -8,6 +8,36 @@ fi
 
 chmod 711 /var/spool/smtpd
 
+# The code of the file_env() function has been borrowed from
+# https://github.com/ix-ai/smtp/blob/master/entrypoint.sh
+# It is licensed under the MIT License, Copyright (c) 2018 Namshi
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "error: both ${var} and ${fileVar} are set (but are mutually exclusive)"
+		exit 1
+	fi
+	local val="$def"
+	if [ "${!var:-}" ]; then
+		val="${!var}"
```

Comment:
I know you just copied the func but still this line makes no sense to me, it will export e.g. `RELAY_USER` env var with the value of the same env var that's already set, what's the point?

Reply:
The idea is to allow to set the value of that env var (given by name) from a file.
As is, it's indeed a noop if the env var is already defined, so the current behavior is kept the same. But it allows to define a variable, say RELAY_USER_FILE with a file name in which the actual content of the RELAY_USER should be read from.

Doing so, I can use opensmtpd easily in a docker swarm cluster using 'docker secret' to store the password: the RELAY_USER en var is normally  defined in the compose file, and I set the RELAY_PASSWORD_FILE to match the docker secret (aka `/run/secrets/<docker_secret_name>`) 

The expression "${!XXX}" will resolve the env  var name, so XXX must exists and be the *name* of another variable. For example:
```
$ TOTO=OTHER_VAR
$ OTHER_VAR=42
$ echo "${!TOTO}"
42
```
So here, we initialize the return value (val) to the default value (if any) (`	local val="$def"`).
Then, if the variable *name* stored in `var` resolve in a value, we use it as return value.  Remember here, `var` is the variable name (e.g. "RELAY_PASSWORD"), not the password itself (i.e. not "$RELAY_PASSWORD").
Else, if the variable name `fileVar` is the *name* of an existing variable which contains a file name (i.e. "$fileVar" is 'RELAY_PASSWORD_FILE', so "${!fileVar}" is actually "$RELAY_PASSWORD_FILE") then we load the value of the returned value from that file.
Then we set the variable named "$var" (RELAY_PASSWORD here) using this return value.

It looks to make sense to me, albeit variable indirect expansion in bash making it not very readable nor sensible.